datafusion/sql/src/planner.rs

```diff
@@ -661,7 +659,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 .iter()
                 .any(|x| x.option == ColumnOption::Null);
             fields.push(Field::new(
-                &normalize_ident(&column.name),
+                &normalize_ident(column.name),
```

Comment:
This is unfortunate to simply drop the `String` immediately, but `Field::new` requires a `&str` (it can't take the String). Filed XX upstream: https://github.com/apache/arrow-rs/pull/3288/files

Reply:
Agree, Look like we can do improvement like above `SubqueryAlias::try_new()`.